### DISCUSSION TITLE
What entitlements do I need for in app purchasing on a macOS app?

### DISCUSSION THREAD

**Discussion by danielkotzer** [airsdk/Adobe-Runtime-Support#2623] [General]
Hi,

Just successfully uploaded my app with the ane's to the mac store on iTunes.
but when the buy now pop up window is supposed to show itself, I get a connection problem error message, which is what 
happens when InAppBillingEvent.SETUP_SUCCESS event was not fired.

Basically I didn't make any changes to the in app purchases system I'm using on iTunes for iPhone devices, the systems are supposed to work the same, 
on iTunes it is the same app page with the same list of in app products. one implementation of the app is for ios app, and one for macos app, but they are working
on the same platform, so why is the ios app working and the macos not working? did I not sign the ane's?
here is the content of my *.command file which I double click to sign the app after it is created on mac:

---------------------------
#!/bin/bash
cd "$(dirname "$BASH_SOURCE")" || {
    echo "Error getting script directory" >&2
    exit 1
}
codesign --entitlements app.entitlements -f --options runtime --strict --timestamp -v -s "3rd Party Mac Developer Application: company name, Inc. (xxxxxxxx)" --deep "app [name.app](http://name.app/)"
-------------------------------------------

the --deep parameter should be taking care of signing everything within the app, even the inner components.

Another problem is that I don't know who to test the app
I know who to do it on iPhone, you build the ipa file with a developer provision profile, and create a testing account 
but on Mac how do I test?
when uploading the mac app I get an error that since I didn't add a provision profile to the app I can't do testflight with it
I contacted Air developers support and they said there is no way to add a provision profile to a mac app at this time, they might add it in the future.
So how do I do debugging, with an app that is "Ready for sale" on iTunes on the Mac store?

maybe I'm missing an entitlement entry in the entitlement list which enables communication with the mac store?
maybe I'm missing an entry in the Info.plist file which enables communication with the mac store?

When I first succeeded creating and uploading my app to the mac store, it was rejected because it didn't work, because it didn't have access to the internet
I asked apple maybe I have to add a line to the entitlements file to access the web, they gave me this line:
        <key>com.apple.security.network.client</key>
        <true/>
When I added this line I got access to the net, the problem was solved and my app was accepted
but I see the in-app purchasing is not working, so it might be that another entitlement is missing.

I saw that if you create the macos app on xCode you have to add in "capabilities" the option of in app purchasing.
How do I add this when building the app with ADT?

**Comment by marchbold** [comment]
There should be no entitlements or info additions required for IAP within the AppStore (there are some for external links for account creation / purchases). 

To add entitlements add the following to your app descriptor:

```xml
<macOS>
    <Entitlements>
        <![CDATA[
            <key>com.apple.security.network.client</key>
            <true/>
        ]]>
    </Entitlements>
</macOS>
```

You should be able to test locally, just make sure you have added all your macos devices as test devices. Testing with testflight I believe will require some additional resigning with a provisioning profile.   

I haven't done this in a while so will have to look into the exact process again.

**Reply by danielkotzer** [reply]
Any progress?
